dist/common/scripts/scylla_util.py

```diff
@@ -154,20 +154,22 @@ def __instance_metadata(self, path, recursive=False):
         return curl(self.META_DATA_BASE_URL + path + "?recursive=%s" % str(recursive).lower(),
                     headers={"Metadata-Flavor": "Google"})
 
+    def is_in_root_devs(self, x, root_devs):
+        for root_dev in root_devs:
+            if root_dev.startswith(os.path.join("/dev/", x)):
+                return True
+        return False
```

Comment:
I don't understand what this is calculating.

Say x is `sda` and `root_devs` is `[/dev/sda1, /dev/sdb1]`. The logic will return True. But what does it mean?

Mixing plain names with paths is confusing, and using startswith() on file names is liable to get wrong results if two devices have names that partially match.

Reply:
@tarzanek Can we change `_non_root_nvmes` to use `glob()` instead of `os.listdir()` to get the full path as follows (the regex also needs tweaking a bit):

```python
nvmes_present = list(filter(nvme_re.match, glob(os.path.join("/dev", "*"))))
```

We can then simplify `is_in_root_devs` to be:

```
def is_in_root_devs(self, x, root_devs):
    return x in root_devs
```

Reply:
@avikivity At high-level, the `_non_root_nvmes` function is attempting to provide a list of NVMe device files that are *not* mounted to "/". I agree that the code is confusing and bit fragile.

Reply:
will fix and retest